src/Update/Nix/FetchGit/Types.hs

```diff
@@ -0,0 +1,57 @@
+{-# LANGUAGE DeriveDataTypeable #-}
+{-# LANGUAGE DeriveTraversable #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Update.Nix.FetchGit.Types where
+
+import           Data.Data             (Data)
+import           Data.Functor.Compose
+import           Data.Text             (Text)
+import qualified Data.Time             (Day)
+import           Data.Typeable         (Typeable)
+import           Nix.Expr              (NExprLoc)
+
+-- TODO: Remove when using base 4.9, necessary at the moment because Compose
+-- doesn't have a data declaration
+deriving instance (Typeable f, Typeable g, Typeable a, Data (f (g a)))
+  => Data (Compose f g a)
+
+-- | The day portion of a date, with no timezone information.
+type Day = Data.Time.Day
```

Comment:
Hm, is there a way to do that without making a list of everything exported from the module?  I don't know what the syntax would be.